### DISCUSSION TITLE
Preventing SaltStack from Overwriting Customized Files in Managed Directories

### DISCUSSION THREAD

**Discussion by BreakChain01** [Security-Onion-Solutions/securityonion#13513] [Show and tell]
In a scenario where SaltStack manages system files, there may be situations where you want to customize specific files without SaltStack overwriting your changes during state applications. This guide demonstrates how to exclude a specific file from being managed by SaltStack while still allowing the rest of the directory to be managed. The solution involves modifying the SaltStack state file to use the `exclude` parameter or directly editing the source file that SaltStack uses for its deployments. This approach ensures that your customizations remain intact across SaltStack operations.

**Comment by BreakChain01** [comment]
I have dmp files. I needed to convert pcap while importing pcap. I made changes to the so-import-pcap file in the following path: /opt/so/saltstack/default/salt/common/tools/sbin_jinja

**Reply by InfosecGoon** [reply]
If you have local customizations that you want to persist through updates, make the changes in /opt/so/saltstack/local/ rather than /opt/so/saltstack/default/.
